Content.Server/GameObjects/Components/Mobs/BuckleComponent.cs

```diff
@@ -0,0 +1,321 @@
+            BuckledTo = null;
+
+            if (Owner.TryGetComponent(out AppearanceComponent appearance))
+            {
+                appearance.SetData(BuckleVisuals.Buckled, false);
+            }
+
+            if (Owner.TryGetComponent(out StunnableComponent stunnable) && stunnable.KnockedDown)
+            {
+                StandingStateHelper.Down(Owner);
+            }
+            else
+            {
+                StandingStateHelper.Standing(Owner);
+            }
+
+            if (Owner.TryGetComponent(out SpeciesComponent species))
+            {
+                species.CurrentDamageState.EnterState(Owner);
+            }
+
+            BuckleStatus();
+
+            return true;
+        }
+
+        public bool ToggleBuckle(IEntity user, IEntity to)
+        {
+            if (BuckledTo == null)
+            {
+                return TryBuckle(user, to);
+            }
+            else if (BuckledTo.Owner == to)
+            {
+                return TryUnbuckle(user);
+            }
+            else
+            {
+                return false;
+            }
+        }
+
+        public override void ExposeData(ObjectSerializer serializer)
+        {
+            base.ExposeData(serializer);
+
+            serializer.DataField(ref _size, "size", 100);
+        }
+
+        protected override void Startup()
+        {
+            base.Startup();
+            BuckleStatus();
+        }
+
+        public override void OnRemove()
+        {
+            base.OnRemove();
+
+            if (BuckledTo != null && BuckledTo.Owner.TryGetComponent(out StrapComponent strap))
+            {
+                strap.Remove(this);
+            }
+
+            BuckledTo = null;
+            BuckleStatus();
+        }
+
+        bool IInteractHand.InteractHand(InteractHandEventArgs eventArgs)
+        {
+            return TryUnbuckle(eventArgs.User);
+        }
+
+        bool IActionBlocker.CanMove()
+        {
+            return BuckledTo == null;
+        }
+
+        bool IActionBlocker.CanChangeDirection()
+        {
+            return BuckledTo == null;
+        }
+
+        bool IEffectBlocker.CanFall()
+        {
+            return BuckledTo == null;
+        }
+
+        [Verb]
+        private sealed class BuckleVerb : Verb<BuckleComponent>
+        {
+            protected override void GetData(IEntity user, BuckleComponent component, VerbData data)
+            {
+                if (!ActionBlockerSystem.CanInteract(user) ||
+                    component.BuckledTo == null)
+                {
+                    data.Visibility = VerbVisibility.Invisible;
+                    return;
+                }
+
+                data.Text = Loc.GetString("Unbuckle");
+            }
+
+            protected override void Activate(IEntity user, BuckleComponent component)
+            {
+                component.TryUnbuckle(user);
+            }
+        }
+    }
+}
```

Comment:
I think it would be best if the buckle verb was a part of the StrapComponent instead. Buckling others can wait on drag and drop, in my opinion.

Reply:
That verb is only for unbuckling others by right clicking them, not for buckling them. I can remove it however if that's also a problem.